### PR TITLE
dumpling: fix dump failed when sequence exists (#30164)

### LINKED ISSUE #30149: dumpling: dump failed when a sequence created
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
tidb> use test;
tidb> create sequence seq;

./dumpling -h host -u test_user -p test_password -P 4000 --filetype sql --threads 32 -o .

```

return dump failed. Since many people do not know what seq is, and error info is `Error 1051: Unknown table ''`, it will make them confused and worried.

### 2. What did you expect to see? (Required)
dump successfully.

### 3. What did you see instead (Required)

```
dump failed: sql: SELECT * FROM `test`.`seq` LIMIT 1: Error 1051: Unknown table ''
```

### 4. What is your TiDB version? (Required)
master

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #30164 to release-5.3
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In tidb repo:
git pr https://github.com/pingcap/tidb/pull/30877
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/tidb.git pr/30877:release-5.3-9d307b00ae86
```

---

### What problem does this PR solve?

Issue Number: close #30149 
Problem Summary:

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dumpling/export/sql_test.go [hunk 1]
```diff
@@ -1736,6 +1736,60 @@ func TestPickupPossibleField(t *testing.T) {
 	}
 }
 
+<<<<<<< HEAD
+=======
```
Comment (kennytm) [inline dumpling/export/sql_test.go:1740]: 👀

**Review by lichunzhu — APPROVED** [review]

**Review by sylzd — APPROVED** [review]

**Pull request merged**